### PR TITLE
Fix/issue 5903

### PR DESCRIPTION
**Pull request by rajyan**
fixes https://github.com/phpstan/phpstan/issues/5903

### CODE REVIEW

File: src/Analyser/NodeScopeResolver.php [hunk 1]
```diff
@@ -823,6 +823,9 @@ private function processStmtNode(
 			if (!$isIterableAtLeastOnce->no()) {
 				$throwPoints = array_merge($throwPoints, $finalScopeResult->getThrowPoints());
 			}
+			if (!$scope->getType($stmt->expr)->isArray()->yes()) {
```
Comment (ondrejmirtes) [inline src/Analyser/NodeScopeResolver.php:826]: What about this instead?

```php
!(new ObjectType(\ArrayAccess))->isSuperTypeOf($scope->getType($stmt->expr)-)->no()
```

Comment (ondrejmirtes) [reply src/Analyser/NodeScopeResolver.php:826]: This will make it not report a false positive for example `array|null`.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Analyser/NodeScopeResolver.php (hunk 1 above)
Comment (rajyan) [reply src/Analyser/NodeScopeResolver.php:826]: Thank you! I'll look into it.

**Comment by ondrejmirtes** [comment]
Sorry, I used a wrong type in the review and fixed it myself :) https://github.com/phpstan/phpstan-src/pull/1051/commits/83e4284e10f7612eaa59852f495aba5882e748ad (ArrayAccess has nothing to do with iteration).

Thank you.

**Review by ondrejmirtes — APPROVED** [review]

**Comment by rajyan** [comment]
@ondrejmirtes 
> ArrayAccess has nothing to do with iteration

Oh, you are right. Sorry that I couldn't notice it. I’m wondering why did the tests passed in local.

**Comment by ondrejmirtes** [comment]
Because these objects **might** implement ArrayAccess in a subclass :) (They're not final.) So the assumption an exception might be thrown is still correct.

**Comment by rajyan** [comment]
I see. Thank you for clarification.

**Pull request merged**